server/data/data-library/src/main/java/org/apache/james/rrt/lib/CanSendFromImpl.java

```diff
@@ -79,29 +84,48 @@ private boolean emailIsAnAliasOfTheConnectedUser(Username connectedUser, Usernam
             .anyMatch(alias -> alias.equals(connectedUser));
     }
 
-    private Stream<Username> relatedAliases(Username user) throws RecipientRewriteTableException {
-        return Stream.concat(
-            Stream.of(user),
+    private Stream<Username> relatedAliases(Username user) {
+        return recursivelyResolve(user, Throwing.<Username, Stream<Username>>function(targetUser ->
             recipientRewriteTable
```

Comment:
What about using a cache to avoid querying the table everytime ? In the scope of each `allValidFromAddressesForUser()` call. I know with the depth being 2, feel like over engineering things.